aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -93,18 +103,57 @@ public void configureAndInitializeInLocalMode() throws AmplifyException {
-        AmplifyConfiguration configuration = AmplifyConfiguration.fromJson(amplifyJson);
-        Amplify.configure(configuration, context);
-
-        JSONObject pluginJson = new JSONObject()
+        ApiCategory mockApiCategory = mockApiCategoryWithGraphQlApi();
+        JSONObject dataStorePluginJson = new JSONObject()
             .put("syncIntervalInMinutes", 60);
-        awsDataStorePlugin.configure(pluginJson, context);
+        AWSDataStorePlugin awsDataStorePlugin = new AWSDataStorePlugin(modelProvider, mockApiCategory);
+        awsDataStorePlugin.configure(dataStorePluginJson, context);
         awsDataStorePlugin.initialize(context);
         assertSyncProcessorStarted();
     }
 
+    /**
+     * Simulate a situation where the user has added the API plugin, but it's
+     * either not pushed or exceptions occur while trying to start up the sync processes.
+     * The outcome is that the local store should still be available and the
+     * host app should not crash.
+     * @throws JSONException If an exception occurs while building the JSON configuration.
+     * @throws AmplifyException If an exception occurs setting up the mock API
+     */
+    @Test
+    public void configureAndInitializeInApiModeWithoutApi() throws JSONException, AmplifyException {
+        ApiCategory mockApiCategory = mockApiPluginWithExceptions();
+        JSONObject dataStorePluginJson = new JSONObject()
+            .put("syncIntervalInMinutes", 60);
+        AWSDataStorePlugin awsDataStorePlugin = new AWSDataStorePlugin(modelProvider, mockApiCategory);
+        awsDataStorePlugin.configure(dataStorePluginJson, context);
+        awsDataStorePlugin.initialize(context);
+
+        // Trick the DataStore since it's not getting initialized as part of the Amplify.initialize call chain
+        Amplify.Hub.publish(HubChannel.DATASTORE, HubEvent.create(InitializationStatus.SUCCEEDED));
+
+        Person person1 = createPerson("Test", "Dummy I");
+        Completable.fromSingle(single -> { // Save a record to local store
+            awsDataStorePlugin.save(person1, itemSaved -> {
+                assertNotNull(itemSaved.item().getId());
+                assertEquals(person1.getLastName(), itemSaved.item().getLastName());
+                single.onSuccess(true);
+            }, single::onError);
+        }).andThen(
+            Completable.fromSingle(single -> { // Verify the record has been saved
+                awsDataStorePlugin.query(Person.class, results -> {
+                    Person actualPerson = results.next();
+                    assertNotNull(actualPerson);
+                    assertFalse(results.hasNext()); // We should only have one result.
+                    assertEquals(person1, actualPerson);
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).doOnError(error -> {
+            fail(error.getMessage());
+        }).blockingAwait();
```

Comment:
```suggestion
        }).blockingAwait(OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
```
^ This won't quite work via the "accept a change" UX, unfortunately.

Anyway, this will help prevent our test from hanging indefinitely in case something goes wrong.